libraries/common/helpers/config/theme.js

```diff
@@ -20,14 +20,15 @@ export function getThemeConfig(appConfig) {
     return themeConfig;
   }
 
-  const { colors = {}, theme: { styles: { globals = {} } = {} } = {} } = appConfig;
+  const { colors = {}, theme = {} } = appConfig;
 
   const oldTheme = process.env.THEME_CONFIG || defaultConfig;
 
   return {
-    font: globals.font,
+    ...theme,
```

Comment:
I'm not sure if it is a good idea to create a completely different config object which diverges from the appConfig.
Externals could actually be surprised to see something else in the returned object than what `appConfig.theme` holds.
This makes injecting config values (from extensions) harder (and actually impossible as is) and for externals it's unclear which one to use now.
I think it would be better to add the correct structure by mutating the original config to the desired structure and then just return the theme property of it. That way the configs stay consistent to each other and maintains simple usage.
In that case `addThemeConfig` or a similar name would be better as well.

Reply:
This new config object is only additional to stay backward compatible with all the thousands of places in the source code that use the "old" theme config from the former `theme-config.js` in the theme's root. Again, this is additional for the time being and will later go away. The alternative would be to refactor all the different places where this is used and that is way out of scope!

Reply:
I think you misunderstood what I meant. I had backwards compatibility in mind.
I'll accept this for now, because there is a related ticket which is affected by this.